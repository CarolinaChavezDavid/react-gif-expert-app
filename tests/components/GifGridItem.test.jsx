import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";

describe('Test in <GifGridItem />', () => { 
    const title = 'Naruto';
    const url = 'https://aws.amazon.com/what-is/batch-processing/';

    test('must match with snapshot', () => { 
        const { container } = render(<GifGridItem title={ title } url= { url } />)
        expect( container ).toMatchSnapshot();

     });

     test('should show the img with right URL and ALT', () => { 
        render(<GifGridItem title={ title } url= { url } />);
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );
        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(url);
        expect(alt).toBe(title);
     });

     test('should show the title in the component', () => { 
        render(<GifGridItem title={ title } url= { url } />);
        expect(screen.getByText(title)).toBeTruthy();
      })

 });