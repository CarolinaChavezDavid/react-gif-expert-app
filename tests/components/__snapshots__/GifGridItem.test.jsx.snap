// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGridItem /> must match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naruto"
      src="https://aws.amazon.com/what-is/batch-processing/"
    />
    <p>
      Naruto
    </p>
  </div>
</div>
`;
